src/sys/stat.rs

```diff
@@ -84,15 +74,15 @@ pub fn makedev(major: u64, minor: u64) -> dev_t {
 }
 
 pub fn umask(mode: Mode) -> Mode {
-    let prev = unsafe { ffi::umask(mode.bits() as mode_t) };
+    let prev = unsafe { libc::umask(mode.bits() as mode_t) };
     Mode::from_bits(prev).expect("[BUG] umask returned invalid Mode")
 }
 
 pub fn stat<P: ?Sized + NixPath>(path: &P) -> Result<FileStat> {
     let mut dst = unsafe { mem::uninitialized() };
     let res = try!(path.with_nix_path(|cstr| {
         unsafe {
-            ffi::stat(cstr.as_ptr(), &mut dst as *mut FileStat)
+            libc::stat(cstr.as_ptr(), &mut dst as *mut FileStat)
```

Comment:
Please either pull the changes to use `libc::stat/fstat` into a separate commit or change the name of this commit as this doesn't just modify `mknod`/`umask`.